### PR TITLE
Simplify check for unit vector in Rotation

### PR DESCRIPTION
**Pull request by xbe**

### CODE REVIEW

**Comment by xbe** [comment]
Check this out, @qcc4cp.

File: src/lib/ops.py [hunk 1]
```diff
@@ -194,8 +194,9 @@ def U1(lam, d=1):
 def Rotation(v: np.ndarray, theta: float) -> np.ndarray:
   """Produce the single-qubit rotation operator."""
 
-  v = np.array(v)
-  if v.shape != (3,) or not np.isclose(a @ a, 1) or not np.all(np.isreal(v)):
+  v = np.asarray(v)
+  if (v.shape != (3,) or not np.isclose(a @ a, 1) or
```
Comment (qcc4cp) [inline src/lib/ops.py:198]: Same here. v @ v instead of a @ a? @xbe

**Comment by qcc4cp** [comment]
Looks good, especially the asarray. But I think this should read v @ v, not a @ a.

File: src/lib/ops.py (hunk 1 above)
Comment (xbe) [reply src/lib/ops.py]: Fixed. The perils of not having automated testing.

File: src/lib/ops.py [hunk 2]
```diff
@@ -192,7 +192,7 @@ def Rotation(v, theta):
   """Produce the single-qubit rotation operator."""
 
   v = np.array(v)
-  if v.shape != (3,) or abs(v.dot(v) - 1.0) > 1e-8 or not np.all(np.isreal(v)):
+  if v.shape != (3,) or not np.isclose(a @ a, 1) or not np.all(np.isreal(v)):
```
[the remark this answers is not in the document]
Comment (xbe) [reply src/lib/ops.py]: Fixed.

**Comment by qcc4cp** [comment]
Thanks

**Pull request merged**